### PR TITLE
Added End Meeting menu for moderators

### LINKED ISSUE #6124: Update logout options for HTML5 client
Currently, there are three options when a moderator logs out of the HTML5 client

image

However, it isn't clear to the user that the 3rd option is clickable.

### LINKED ISSUE #6245: Add "End Meeting" as a dropdown menu item

### LINKED ISSUE #6542: Downgraded user can still end the meeting
When a moderator downgrades a user to viewer, they are still able to end the meeting upon logout.

### PR DESCRIPTION
**Pull request by antobinary**
- added a new item in the navbar dropdown 'End Meeting' available only to moderators
- added a confirmation modal for when End Meeting is triggered
- removed the confirmation modal for Logout menu item
- split RecordingIndicator into view for moderator (with actions) and for attendee - readOnly
- added a reactive check for both EndMeeting and Recording trigger so we can properly update the UI if a user is promoted/demoted
- rewrote the settings-dropdown component to avoid old `componentWillReceiveProps` method. Given that we have dynamically omittable [fullscreen, help, end meeting and hotkeys] menu items, I had to deviate from single-time constructed array of dropdown items and moved the items to be re-rended on settings-dropdown open/close.
Closes #6124 
Closes #6245 
Closes #6542

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/nav-bar/settings-dropdown/component.jsx [hunk 1]
```diff
@@ -209,10 +228,12 @@ class SettingsDropdown extends PureComponent {
       shortcuts: OPEN_OPTIONS_AK,
     } = this.props;
 
+    const { isSettingOpen } = this.state;
+
     return (
       <Dropdown
         autoFocus
-        isOpen={this.state.isSettingOpen}
+        isOpen={isSettingOpen}
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/nav-bar/settings-dropdown/component.jsx]: I think this prop was renamed.
https://github.com/bigbluebutton/bigbluebutton/commit/8564e53ffb164ca991032ea4e6c03d8300faf50a#diff-5f1507eaefcaf30f75ce7be57afbf439R104

File: bigbluebutton-html5/imports/ui/components/end-meeting-confirmation/component.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      <Modal
+        overlayClassName={styles.overlay}
+        className={styles.modal}
+        onRequestClose={this.closeModal}
+        hideBorder
+      >
+        <div className={styles.container}>
+          <div className={styles.header}>
+            <div className={styles.title}>
+              {intl.formatMessage(intlMessages.endMeetingTitle)}
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/end-meeting-confirmation/component.jsx]: modal component has title prop to pass it

File: bigbluebutton-html5/imports/ui/components/nav-bar/settings-dropdown/component.jsx (hunk 1 above)
Comment (antobinary) [reply bigbluebutton-html5/imports/ui/components/nav-bar/settings-dropdown/component.jsx]: Could you please confirm that it has to be `isOpen` now?

File: bigbluebutton-html5/imports/ui/components/end-meeting-confirmation/component.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+class EndMeetingComponent extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    const {
+      closeModal,
+      endMeeting,
+    } = props;
+
+    this.closeModal = closeModal;
```
[the remark this answers is not in the document]
Comment (antobinary) [reply bigbluebutton-html5/imports/ui/components/end-meeting-confirmation/component.jsx]: The example I was looking at had it this way :D I confirmed this is unnecessary, removed it and also went back to edit the "example" code I had used. Thanks for pointing this out!

**Pull request merged**